src/flask_pyoidc/flask_pyoidc.py

```diff
@@ -319,3 +333,203 @@ def valid_access_token(self, force_refresh=False):
                        id_token_jwt=response.get('id_token_jwt'),
                        refresh_token=response.get('refresh_token'))
         return access_token
+
+    @staticmethod
+    def _check_authorization_header(request) -> bool:
+        """Look for authorization in request header.
+
+        Parameters
+        ----------
+        request : werkzeug.local.LocalProxy
+            flask request object.
+
+        Returns
+        -------
+        bool
+            True if the request header contains authorization else False.
+        """
+        if 'Authorization' in request.headers and request.headers['Authorization'].startswith('Bearer '):
+            return True
+        return False
+
+    @staticmethod
+    def _parse_access_token(request) -> str:
+        """Parse access token from the authorization request header.
+
+        Parameters
+        ----------
+        request : werkzeug.local.LocalProxy
+            flask request object.
+
+        Returns
+        -------
+        accept_token : str
+            access token from the request header.
+        """
+        _, access_token = request.headers['Authorization'].split(maxsplit=1)
+        return access_token
+
+    def introspect_token(self, request, client, scopes: list = None) -> Union[
+            TokenIntrospectionResponse, bool]:
+        """RFC 7662: Token Introspection
+        The Token Introspection extension defines a mechanism for resource
+        servers to obtain information about access tokens. With this spec,
+        resource servers can check the validity of access tokens, and find out
+        other information such as which user and which scopes are associated
+        with the token.
+
+        Parameters
+        ----------
+        request : werkzeug.local.LocalProxy
+            flask request object.
+        client : flask_pyoidc.pyoidc_facade.PyoidcFacade
+            PyoidcFacade object contains metadata of the provider and client.
+        scopes : list
+            Specify scopes that are bound for the end endpoint.
+
+        Returns
+        -------
+        bool
+            True if access_token is valid else False.
+
+        Raises
+        ------
+        NotForMe
+            If access_token is invalid.
+
+        """
+        received_access_token = self._parse_access_token(request)
+        # send token introspection request
+        result = client._token_introspection_request(
+            access_token=received_access_token)
+        logger.debug(result)
+        # Check if access_token is valid, active can be True or False
+        if not result.get('active'):
+            return False
```

Comment:
How about returning `None` instead (applies to all the early returns in this method)?
I think return type `typing.Optional[TokenIntrospectionResponse]` is more clear.